geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSScanIntegrationTest.java

```diff
@@ -54,295 +72,394 @@ public void tearDown() {
 
-    List<Object> result;
+    ScanResult<byte[]> result = jedis.sscan(KEY_BYTES, ZERO_CURSOR_BYTES, scanParams);
 
-    List<byte[]> allEntries = new ArrayList<>();
-    String cursor = "0";
+    assertThat(result.isCompleteIteration()).isTrue();
+    assertThat(result.getResult()).containsOnly(MEMBER_ONE.getBytes(),
+        MEMBER_TWELVE.getBytes());
+  }
 
-    do {
-      result =
-          (List<Object>) jedis.sendCommand("a", Protocol.Command.SSCAN, "a", cursor, "COUNT", "2",
-              "COUNT", "1");
-      allEntries.addAll((List<byte[]>) result.get(1));
-      cursor = new String((byte[]) result.get(0));
-    } while (!Arrays.equals((byte[]) result.get(0), "0".getBytes()));
+  @Test
+  public void givenSetWithThreeEntriesAndMultipleMatchArguments_returnsOnlyElementsMatchingLastMatchArgument() {
+    jedis.sadd(KEY, MEMBER_ONE, MEMBER_TWELVE, MEMBER_THREE);
+    ScanParams scanParams = new ScanParams().match("3*").match("1*");
+
+    ScanResult<byte[]> result = jedis.sscan(KEY_BYTES, ZERO_CURSOR_BYTES, scanParams);
 
-    assertThat((byte[]) result.get(0)).isEqualTo("0".getBytes());
-    assertThat(allEntries).containsExactlyInAnyOrder("1".getBytes(),
-        "12".getBytes(),
-        "3".getBytes());
+    assertThat(result.getCursor()).isEqualTo(ZERO_CURSOR);
+    assertThat(result.getResult()).containsOnly(MEMBER_ONE.getBytes(),
+        MEMBER_TWELVE.getBytes());
   }
 
   @Test
-  public void givenMatch_returnsAllMatchingMembersWithoutDuplicates() {
-    jedis.sadd("a", "1", "12", "3");
-
+  public void givenLargeCountAndMatch_returnsOnlyMatchingMembers() {
+    Set<byte[]> initialMemberData = initializeThousandMemberByteSet();
     ScanParams scanParams = new ScanParams();
-    scanParams.match("1*");
+    // There are 111 matching members in the set 0..999
+    scanParams.match("9*");
+    // Choose a large COUNT to ensure that some matching members are returned
+    scanParams.count(950);
+
+    ScanResult<byte[]> result = jedis.sscan(KEY_BYTES, ZERO_CURSOR_BYTES, scanParams);
+
+    List<byte[]> returnedMembers = result.getResult();
+    // We know that we must have found at least 61 matching members, given the size of COUNT and the
+    // number of matching members in the set
+    assertThat(returnedMembers.size()).isGreaterThanOrEqualTo(61);
+    assertThat(returnedMembers).isSubsetOf(initialMemberData);
+    assertThat(returnedMembers).allSatisfy(bytes -> assertThat(new String(bytes)).startsWith("9"));
+  }
 
-    ScanResult<byte[]> result =
-        jedis.sscan("a".getBytes(), "0".getBytes(), scanParams);
+  @Test
+  public void givenMultipleCountAndMatch_usesLastSpecified() {
+    Set<byte[]> initialMemberData = initializeThousandMemberByteSet();
+    // Choose a large COUNT to ensure that some matching members are returned
+    // There are 111 matching members in the set 0..999
+    ScanParams scanParams = new ScanParams().count(20).match("1*").count(950).match("9*");
```

Comment:
See the comment above about using `sendCustomSscanCommand()` instead of multiple calls to `ScanParams` methods here.